pangeo_forge_recipes/serialization.py

```diff
@@ -0,0 +1,133 @@
+from collections.abc import Collection
+from dataclasses import asdict
+from enum import Enum
+from hashlib import sha256
+from json import dumps
+from typing import Any, List, Sequence
+
+from .patterns import FilePattern, Index
+
+
+def json_default(obj: Any):
+    """For objects which are not serializable with ``json.dumps``, this function defines
+    type-specific handlers which extract a serializable value from the object.
+
+    :param obj: Any object which is not serializable to ``json``.
+    """
+
+    if isinstance(obj, Enum):  # custom serializer for FileType, CombineOp, etc.
+        return obj.value
+    elif isinstance(obj, FilePattern):
```

Comment:
Here's how you can duck type this to avoid importing FilePattern

```suggestion
    elif hasattr(obj, "sha256"):
```

Reply:
Oh gotcha good idea!